C++/Logger/Logger.cpp

```diff
@@ -7,46 +7,47 @@ namespace Toolbox::Log
 {
 
 Logger::Logger (StringSink & sink)
-: _threshold (Level::TRACE)
+: _default (Level::INFO)
+, _threshold (Level::TRACE)
 , _sink (sink)
 , _stringVisitor ()
 {
   
 }
 
-void Logger::log (const Level & level, const String & message) const
+void Logger::log (const String & message) const
 {
-  logAtLevel (level, message);
+  logIfAboveThreshold (_default, message);
 }
 
 void Logger::trace (const String & message) const
 {
-  logAtLevel (Level::TRACE, message);
+  logIfAboveThreshold (Level::TRACE, message);
 }
 
 void Logger::debug (const String & message) const
 {
-  logAtLevel (Level::DEBUG, message);
+  logIfAboveThreshold (Level::DEBUG, message);
```

Comment:
I chose logIfAboveThreshold for its readability in the code.  Functions are meant to do one thing and you should know what it does without having to dive into the details.  Since logIfAboveThreshold is a private function it doesn't hurt or look weird to users of the interface.